src/AutoBuild.jl

```diff
@@ -1268,80 +1275,98 @@ function build_jll_package(src_name::String, build_version::VersionNumber, code_
-              num_chars = ccall((:count_characters, libdataproc), Cint, (Cstring, Cint), data_lines[1], length(data_lines[1]))
+                        The following JLL packages are required by `$(src_name)_jll.jl`:""")
+            println(io)
+            print_jll.(Ref(io), dependencies)
+        end
+        if length(keys(products_info)) > 0
+            println(io)
+            println(io, """
+                        ## Products
 
-              # For executable products, you can use the exported variable name as a function that you can call
-              mungify_exe() do mungify_exe_path
-                  run(`\$mungify_exe_path \$num_chars`)
-              end
-              ```
-              """)
+                        The code bindings within this package are autogenerated from the following `Products`:
+                        """)
+            for (p, _) in products_info
+                print_product(io, p)
+            end
+        end
     end
 
-    # Generate LICENSE.md
-    open(joinpath(code_dir, "LICENSE.md"), "w") do io
-        print(io, """
-        Copyright $(year(now()))
-
-        Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
-
-        The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-
-        THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-        """)
+    # Add before the license a note about to what files this applies
+    license = strip(read(joinpath(code_dir, "LICENSE"), String))
+    note = "The Julia source code within this repository (all files under `src/`) are released under the terms of the MIT \"Expat\" License, the text of which is included below.  This license does not apply to the binary package wrapped by this Julia package and automatically downloaded by the Julia package manager upon installing this wrapper package.  The binary package's license is shipped alongside the binary itself and can be found within the `share/licenses/$(src_name)` directory within its prefix."
+    # Since this function can be called multiple times, we must make sure that
+    # the note is written only once.  Do nothing it is already there.
```

Comment:
> That seems really brittle.

Why?  It's just how our CI system on Yggdrasil works, which can call the `build_jll_package` function multiple times, there is nothing fundamental.

> Where does the original license come from?

https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/b2cd2e35cb34cea0983c3b5ca88220d98b2d8940/src/AutoBuild.jl#L854

Reply:
With "brittle" I meant that if the content of `note_lines` is changed (e.g. to reformat it ;-) ) then this code would prepend it once again, instead of replacing the old variant.

So, why not just specify the full content of the LICENSE file here, and write it out unconditionally? What is gained by prepending this text? One thing I can think of is that in principle this would allow somebody to manually edit the JLL repository, to replace the license. But is that even supported? Who even would have such access?

Reply:
One point is that the line with the copyright owner is custom, as you noted, I didn't want to change that one

Reply:
But for all JLLs on `JuliaBinaryWrappers` this is just set to `JuliaBinaryWrappers`... That doesn't seem very useful...

BTW, it also always sets the current year as copyright year, which also is not quite right. Ideally it'd be `<year the jll was created> - <current_year>`; folded together into a single line if this is the same year. Of course that might be annoying to implement (I guess one could query git for the creation date of the build file?), so I am not suggestion you waste your time on this -- just thought I should mention it for completeness. :-)

Reply:
We don't currently have any of the information you're suggesting to use

> Ideally it'd be <year the jll was created> - <current_year>; folded together into a single line if this is the same year. Of course that might be annoying to implement (I guess one could query git for the creation date of the build file?)

We must assume that a file must checked in a git repository, which isn't a requirement anywhere.

I think you may want to add more metadata to the builder.

Reply:
Yeah, I realize that this information is not available (which is why I wrote "Ideally it'd be", but I should have made that clearer). And in fact I do think there should be more metadata in the builder.

Anyway, I just thought about it some more, and I now think my "ideally..." suggestion was actually totally *wrong*: this license after all applies to the generated wrapper. And that wrapper is generated by BinaryBuilder. So there is a case to be made that its copyright belongs to the authors of BinaryBuilder resp. the AutoBuild code; the person who run build_tarballs just pressed a button, but the actual creation of the JLL content is not done by them ...

So I think I'd either remove that copyright line, or use something like `Copyright 2019-2020 The authors of the BinaryBuilder.jl package".

Anyway, this is of course not very important, it just irks me a little to know that I might soon see `Copyright (c) 2020 JuliaBinaryWrappers` in lots of LICENSE files -- JuliaBinaryWrappers isn't a legal entity and can't hold any copyrights :-)